### ISSUE TITLE
`coverage xml` shouldn't bail out on parse error

### ISSUE THREAD

**Issue by nedbat** [coveragepy/coveragepy#396]
*Originally reported by* **Ionel Cristian Mărieș (Bitbucket: [ionelmc](https://bitbucket.org/ionelmc), GitHub: [ionelmc](https://github.com/ionelmc))**

----------------------------------------

I have a module that has to use unsupported syntax (it's only imported if you're on the right python).

But now `coverage xml` fails cause of that.

```
$ coverage xml
Couldn't parse '/home/ionel/osp/pytest-benchmark/src/pytest_benchmark/pep418.py' as Python source: 'invalid syntax' at line 47
```

Can we make it a warning instead?

----------------------------------------
- Bitbucket: https://bitbucket.org/ned/coveragepy/issue/396

**Comment by nedbat** [comment]
The ``-i`` flag should work for you here, no?

**Comment by nedbat** [comment]
*Original comment by* **Ionel Cristian Mărieș (Bitbucket: [ionelmc](https://bitbucket.org/ionelmc), GitHub: [ionelmc](https://github.com/ionelmc))**

----------------------------------------

Doh! I don't know how I missed it

**Comment by nedbat** [comment]
*Original comment by* **Ionel Cristian Mărieș (Bitbucket: [ionelmc](https://bitbucket.org/ionelmc), GitHub: [ionelmc](https://github.com/ionelmc))**

----------------------------------------

Doooh:)

**Comment by nedbat** [comment]
*Original comment by* **Ionel Cristian Mărieș (Bitbucket: [ionelmc](https://bitbucket.org/ionelmc), GitHub: [ionelmc](https://github.com/ionelmc))**

----------------------------------------

Actually ... shouldn't it show some warnings if `--ignore-errors` is on?

**Comment by nedbat** [comment]
You mean show the same messages, but as warnings?

**Comment by nedbat** [comment]
*Original comment by* **Ionel Cristian Mărieș (Bitbucket: [ionelmc](https://bitbucket.org/ionelmc), GitHub: [ionelmc](https://github.com/ionelmc))**

----------------------------------------

Yes. A little bit of extra output doesn't hurt and helps spot potential issues.

**Comment by nedbat** [comment]
*Original comment by* **Matthew Boehm (Bitbucket: [mattboehm](https://bitbucket.org/mattboehm), GitHub: [mattboehm](https://github.com/mattboehm))**

----------------------------------------

Looking into this.

**Comment by nedbat** [comment]
*Original comment by* **Matthew Boehm (Bitbucket: [mattboehm](https://bitbucket.org/mattboehm), GitHub: [mattboehm](https://github.com/mattboehm))**

----------------------------------------

Pull request: https://github.com/nedbat/coveragepy/pull/8

**Comment by nedbat** [comment]
Applied and fixed in [551dc23 (bb)](https://bitbucket.org/ned/coveragepy/commits/551dc23).
